libs/ui/src/typography.tsx

```diff
@@ -0,0 +1,234 @@
+  & + h2 {
+    /* stylelint-disable-next-line declaration-no-important */
+    margin-top: 0 !important;
+  }
+`;
+
+const StyledH2 = styled.h2<HeadingProps>`
+  ${headingStyles}
+
+  font-size: ${(p) => p.theme.sizes.headingsRem.h2}rem;
+  font-weight: 600;
+
+  /* Override the top-margin spacing for adjacent next-level headings. */
+  & + h3 {
+    /* stylelint-disable-next-line declaration-no-important */
+    margin-top: 0 !important;
+  }
+`;
+
+const StyledH3 = styled.h3<HeadingProps>`
+  ${headingStyles}
+
+  font-size: ${(p) => p.theme.sizes.headingsRem.h3}rem;
+
+  /* Override the top-margin spacing for adjacent next-level headings. */
+  & + h4 {
+    /* stylelint-disable-next-line declaration-no-important */
+    margin-top: 0 !important;
+  }
+`;
+
+const StyledH4 = styled.h4<HeadingProps>`
+  ${headingStyles}
+
+  font-size: ${(p) => p.theme.sizes.headingsRem.h4}rem;
+
+  /* Override the top-margin spacing for adjacent next-level headings. */
+  & + h5 {
+    /* stylelint-disable-next-line declaration-no-important */
+    margin-top: 0 !important;
+  }
+`;
+
+const StyledH5 = styled.h5<HeadingProps>`
+  ${headingStyles}
+
+  font-size: ${(p) => p.theme.sizes.headingsRem.h5}rem;
+
+  /* Override the top-margin spacing for adjacent next-level headings. */
+  & + h6 {
+    /* stylelint-disable-next-line declaration-no-important */
+    margin-top: 0 !important;
+  }
+`;
+
+const StyledH6 = styled.h6<HeadingProps>`
+  ${headingStyles}
+
+  font-size: ${(p) => p.theme.sizes.headingsRem.h6}rem;
+  line-height: ${(p) => p.theme.sizes.lineHeight * 0.85};
+  font-weight: 600;
+`;
+
+/** Adjusts the font styles for a given piece of inline text. */
+export function Font(props: FontProps): JSX.Element {
+  return <StyledFont {...props} />;
+}
+
+/** Styled h1 heading. */
+export function H1(props: HeadingProps): JSX.Element {
+  return <StyledH1 {...props} />;
+}
+
+/** Styled h2 heading. */
+export function H2(props: HeadingProps): JSX.Element {
+  return <StyledH2 {...props} />;
+}
+
+/** Styled h3 heading. */
+export function H3(props: HeadingProps): JSX.Element {
+  return <StyledH3 {...props} />;
+}
+
+/** Styled h4 heading. */
+export function H4(props: HeadingProps): JSX.Element {
+  return <StyledH4 {...props} />;
+}
+
+/** Styled h5 heading. */
+export function H5(props: HeadingProps): JSX.Element {
+  return <StyledH5 {...props} />;
+}
+
+/** Styled h6 heading. */
+export function H6(props: HeadingProps): JSX.Element {
+  return <StyledH6 {...props} />;
+}
+
+/** Styled block paragraph text for regular copy. */
+export function P(props: FontProps): JSX.Element {
+  return <StyledP {...props} />;
+}
+
+/** Styled block caption for small sub-text copy. */
+export function Caption(props: FontProps): JSX.Element {
+  return <StyledCaption {...props} />;
+}
+
+/**
+ * Styled block text for pre-formatted copy.
+ * Sanitizes and renders HTML text as HTML elements and preserves whitespace and
+ * newline formatting.
+ */
```

Comment:
Is this inspired by certain use cases within the apps?

Reply:
Yeah, noticed there's an existing `TextWithLineBreaks` component used when rendering some pre-formatted text in election definitions (I think it's text configured by election officials)